docs/autorenew-feature.md

```diff
@@ -0,0 +1,71 @@
+# The autorenew feature
+
+## Goal
+-	Allow Hedera Services to automatically search for entities such as accounts, files, smart contracts, topics, tokens... that are about to expire and renew them.
+
+## Design
+- Introduce new settings in `application.properties`:
+  * `autorenew.isEnabled`
+  * `autorenew.numberOfEntitiesToCheck`
+  * `autorenew.maxNumberOfEntitiesToRenew`
+  * `autorenew.gracePeriod`
+- Each Hedera entity has an `expirationTime` which is the effective consensus timestamp at (and after) which the entity is set to expire.
+- Each Hedera entity also has an `autoRenewAccount` which is the account to pay for the fee at renewal. This `autoRenewAccount` could be itself if the entity is a crypto account or an account associated with the entity when it was created.
```

Comment:
Contracts don't have `autoRenewAccount` and `autoRenewPeriod`. Should they be added as well to be consistent?

Reply:
#1194